### PR TITLE
Support Copy with Remote Object Stores in datafusion-cli

### LINKED ISSUE #8907: Support Copy with Remote Object Stores in datafusion-cli
### Is your feature request related to a problem or challenge?

This unfortunately does not work in datafusion-cli currently:
```
DataFusion CLI v34.0.0
❯ copy (values (1,2)) to 's3://dfcli-test-bucket/test3.parquet';
Internal error: No suitable object store found for s3://dfcli-test-bucket/.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
❯ copy (values (1,2)) to 's3://dfcli-test-bucket2/test3.parquet';
Internal error: No suitable object store found for s3://dfcli-test-bucket2/.
This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

But this does:

```
❯ create external table remote_table2(a int, b int) stored as parquet location 's3://dfcli-test-bucket2/';
0 rows in set. Query took 0.001 seconds.

❯ insert into remote_table2 values (1,2);
+-------+
| count |
+-------+
| 1     |
+-------+
1 row in set. Query took 0.272 seconds.

❯ select * from remote_table2;
+---+---+
| a | b |
+---+---+
| 1 | 2 |
+---+---+
```

### Describe the solution you'd like

Support "on the fly" registration of remote object store for copy statements similar to create external table.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by manoj-inukolunu**
## Which issue does this PR close?

Closes #8907 .

## Are these changes tested?
Yes , added tests.

### CODE REVIEW

**Comment by manoj-inukolunu** [comment]
Hello @alamb  , Here is a draft for https://github.com/apache/arrow-datafusion/issues/8907 . I am yet to add tests. Currently `DefaultObjectStoreRegistry`
```
pub fn new() -> Self {
        let object_stores: DashMap<String, Arc<dyn ObjectStore>> = DashMap::new();
        object_stores.insert("file://".to_string(), Arc::new(LocalFileSystem::new()));
        Self { object_stores }
    }
```
 registers only `LocalFileSystem` .
 I am wondering if it should register other stores as well , like S3, GCS etc.

File: datafusion-cli/src/exec.rs [hunk 1]
```diff
@@ -260,6 +256,36 @@ async fn exec_and_print(
     Ok(())
 }
 
+async fn create_plan(
```
Comment (alamb) [inline datafusion-cli/src/exec.rs:259]: This makes sense to me. Thank you @manoj-inukolunu

**Review by alamb — APPROVED** [review]
Thank you @manoj-inukolunu  -- this makes sense to me and is a nice improvement

**Comment by alamb** [comment]
BTW I filed https://github.com/apache/arrow-datafusion/issues/9133 for a small follow on too

**Comment by alamb** [comment]
Thanks again @manoj-inukolunu

**Pull request merged**